cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,598 @@
+                "Select": Checkbox(locator=".//input"),
+                1: Text('.//span/button[contains(@class,"btn btn-link")]'),
+            },
+        )
+        filter_by_dropdown = SelectorDropdown("id", "filterFieldTypeMenu_vms_step")
+        search_box = SearchBox(
+            locator='.//div[contains(@class, "modal-content")]'
+            '//div[contains(@class,"input-group")]/input'
+        )
+        clear_filters = Text(".//a[text()='Clear All Filters']")
+        error_text = Text(
+            './/h3[contains(@class,"blank-slate-pf-main-action") and '
+            'contains(text(), "Error:")]'
+        )
+        error_icon = Text(locator='.//span[contains(@class, "pficon-error-circle-o")]')
+        popover_text = Text(locator='.//div[contains(@class, "popover-content")]')
+
+        @property
+        def is_displayed(self):
+            return self.table.is_displayed and (
+                len(self.browser.elements(".//div[contains(@class,'spinner')]")) == 0
+            )
+
+        def filter(self, type, name):
+            try:
+                self.filter_by_dropdown.item_select(type)
+            except NoSuchElementException:
+                logger.info("{} not present in filter dropdown!".format(type))
+            self.search_box.fill(name)
+
+        def select_by_name(self, vm_name):
+            self.filter("VM Name", vm_name)
+            vms_selected = []
+            for row in self.table.rows():
+                if vm_name in row.read()["VM Name"]:
+                    row.select.fill(True)
+                    vms_selected.append(row.read()["VM Name"])
+            return vms_selected
+
+    @View.nested
+    class advanced(View):  # noqa
+        pre_playbook = BootstrapSelect("preMigrationPlaybook")
+        post_playbook = BootstrapSelect("postMigrationPlaybook")
+        pre_checkbox = Text(locator='.//input[contains(@id, "pre_migration_select_all")]')
+        post_checkbox = Text(locator='.//input[contains(@id, "post_migration_select_all")]')
+
+    @View.nested
+    class options(View):  # noqa
+        create = Button("Create")
+        run_migration = RadioGroup('.//div[contains(@id,"migration_plan_choice_radio")]')
+
+        @property
+        def is_displayed(self):
+            return self.run_migration.is_displayed
+
+    @View.nested
+    class results(View):  # noqa
+        close = Button("Close")
+        msg = Text('.//h3[contains(@id,"migration-plan-results-message")]')
+
+    @property
+    def is_displayed(self):
+        return self.title.text == self.form_title_text
+
+
+class MigrationPlanRequestDetailsView(View):
+    migration_request_details_list = MigrationPlanRequestDetailsList("plan-request-details-list")
+    sort_type = SelectorDropdown("id", "sortTypeMenu")
+    paginator_view = View.include(V2vPaginatorPane, use_parent=True)
+    search_box = SearchBox(locator=".//div[contains(@class,'input-group')]/input")
+    clear_filters = Text(".//a[text()='Clear All Filters']")
```

Comment:
Question:
I am suspect this  should be part of `SearchBox`? can you please help me to understand this implementation.

Reply:
Clear All Filters is a link (href) to clear all filters .